aws/src/database/dynamodb/scripts/contact_details_table/scan_and_patch_new_phone_numbers.py

```diff
@@ -0,0 +1,81 @@
+from dataclasses import dataclass
+
+from mypy_boto3_dynamodb.service_resource import Table
+
+#from aws.src.database.dynamodb.utils.get_by_secondary_index import get_by_secondary_index
+from aws.src.database.dynamodb.utils.get_dynamodb_table import get_dynamodb_table
+from aws.src.utils.csv_to_dict_list import csv_to_array
+from aws.src.utils.logger import Logger
+from aws.src.utils.progress_bar import ProgressBar
+from enums.enums import Stage
+from decimal import Decimal
+import time
+#
+# IMPORTANT:
+#
+# This script uses the scan() operation in DyanmoDB which is very expensive
+# if used repeatedly. DO NOT invoke these methods in a loop. Rather, supply
+# a list of affected records and run the script once
+#
+
+@dataclass
+class Config:
+    TABLE_NAME = "ContactDetails"
+    LOGGER = Logger()
+    STAGE = Stage.HOUSING_DEVELOPMENT
+    FILE_PATH = "aws\src\database\input\\bad_phone_numbers.csv"
+    HEADING_FILTERS = {
+        "id": lambda x: bool(x),
+    }
+
+total_count = 0
+update_count = 0
+
+def process_item(item, bad_phone_numbers, dynamo_table, logger):
+    global update_count
+    partition_key = item['targetId']
+    sort_key = item['id']
+    #logger.log(f"Process Item {partition_key} / {sort_key}")
+    if not item.get('contactInformation'):
+        return
+    if not item.get('contactInformation').get('value'):
+        return
+    if item.get('contactInformation').get('contactType') != 'phone':
+        return
+    phone = item.get('contactInformation').get('value')
+    if phone in bad_phone_numbers:
+        logger.log(f"Item {partition_key} / {sort_key} has a bad phone number! Erasing....")
+        item["contactInformation"]["value"] = ''
+        dynamo_table.put_item(Item=item)
+        update_count += 1
+
+def process_scan(scan_results, bad_phone_numbers, dynamo_table, logger):
+    global total_count
+    global update_count
```

Comment:
The script would work without any of the global statements as the variables are already file (i.e. "module") scoped on lines 31-32. The global keyword could take function-scoped variables and make them global.

If you want to be more explicit about which global variables are being used by the function, you could pass them in as function arguments